freshmaker/pyxis.py

```diff
@@ -215,6 +215,129 @@ def get_images_by_digests(self, digests):
+            dict containing nvrs of bundle images from that repo and
+            auto_rebuild tags of that repo
+        :return: pairs of (registry, repository) of every repo with images ever
+            published to a repo with auto_rebuild tag
+        :rtype set(tuple(str, str)):
+        """
+        params = {'include': 'data.brew.build'}
+        ret_repos = set()
+        for (reg, repo), info in reg_repo_info.items():
+            if not info.get('nvrs'):
+                continue
+            fltr = f"brew.build=in=({','.join(info['nvrs'])})"
+            params["filter"] = fltr
+            for tag in info.get('tags', []):
+                tag_history = \
+                    self._pagination(f"repositories/registry/{reg}/"
+                                     f"repository/{repo}/tag/{tag}", params)
+                # If there is at least one record with nvr(brew.build) of the
+                # image, it means that image was published to a repo with
+                # auto_rebuild tag
+                if tag_history:
+                    ret_repos.add((reg, repo))
+                    break
+        return ret_repos
+
+    def get_auto_rebuild_taged_images(self, bundle_images):
+        """
+        Determine which bundle images are part of container repository and tag
+        that has auto-rebuilds enabled
+
+        :param list bundle_images: Images of operator bundles that should
+            be filtered
+        :return: Images that where published to repository with auto_rebuild tag
+        :rtype: list
+        """
+        reg_repo_info = {}
+        ret_images = []
+        for bundle_image in bundle_images:
+            bundle_nvr = bundle_image.get('brew', {}).get('build')
+            if not bundle_nvr:
+                log.warning("One of bundle images doesn't have brew.build")
+                continue
+            if not bundle_image.get('repositories'):
+                log.warning("Bundle image %s doesn't have repositories set",
+                            bundle_nvr)
+                continue
+            # construct mapping of (registry, repository) -> {'nvrs': {bundles_nvrs}}
+            for repo in bundle_image.get('repositories'):
+                reg_repo = (repo.get('registry'), repo.get('repository'))
+                if reg_repo not in reg_repo_info:
+                    reg_repo_info[reg_repo] = {'nvrs': {bundle_nvr}}
+                elif 'nvrs' not in reg_repo_info[reg_repo]:
+                    reg_repo_info[reg_repo]['nvrs'] = {bundle_nvr}
+                else:
+                    reg_repo_info[reg_repo]['nvrs'].add(bundle_nvr)
+
+        # Add auto_rebuild_tags to info structures for every repo
+        self._add_repositories_info(reg_repo_info)
+        # Get tag history for every repo and filter out repos that didn't have
+        # bundle images in their history
+        repos = self._filter_auto_rebuild_repos(reg_repo_info)
```

Comment:
If I understand the code correctly, this is returning all the repos which have at least one vulnerable bundle image that was at least once tagged with an auto rebuild tag. Later on, the code determines which bundle images that are published to these repos to then rebuild. This is not sufficient since it's possible one of the vulnerable bundle images was tagged to "registry/ns/repo:v4.5" but auto rebuilds are only set to the `v4.6` tag on that repo.

Reply:
@mprahl  In that case will it be enough to check in addition if at least one image tag(that **repositories->tags->name**) is in **auto_rebuild_tags** list of the repo? I will push changes so you can take a look.

Reply:
@apaplaus I think what you'll need to do is map the NVRs of the bundle images to the repositories like you are doing. Then check the tag history of each autorebuild tag to see of any of the NVRs on that repository were or are currently tagged with one of the autorebuild tags.

Reply:
I reread your comment and I think I originally misunderstood. Let me check the code to see.

Reply:
@apaplaus I think you could modify `_filter_auto_rebuild_repos` to create and return a set of all the NVRs that were once tagged with an autorebuild tag. I think in the section that you do `if tag_history:`, you could just loop through the NVRs returned and add them to the set since we know that the NVRs are limited to just those that are bundle images of interest. Then the additional for loop of `for bundle_image in bundle_images:` in this method could be removed and it could just return the set of NVRs.